synonyms.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import "errors"
+
+type Graph map[string]map[string]bool
```

Comment:
This really isn't the most intuitive data type for that kind of problem, but let's see how you use it :)